### ISSUE TITLE
New Customer

### ISSUE THREAD

**Issue by saud-2** [lavaloon-eg/ksa_compliance#86]
After last update, I can't create new customer. when click add customer button nothing pop-up .

**Comment by mhaggag** [comment]
That's odd, the update didn't include any changes to the customer scripts or logic [as far as I can see](https://github.com/lavaloon-eg/ksa_compliance/commit/cee77e3e9a1702a34f11205c37e26f110b71d688)

Is this in the customer list page? Do you get any JavaScript errors in the browser console? (Developer Tools -> Console)

**Comment by saud-2** [comment]
yes in customer list page . this is from console

script_manager.js:32 ƒ (frm){
    frm.set_df_property('custom_additional_ids', 'cannot_delete_rows', 1);
    frm.set_df_property('custom_additional_ids', 'cannot_add_rows', 1);
  }

field_group.js:199 Uncaught (in promise) 
TypeError: Cannot read properties of undefined (reading 'df')
    at frappe.ui.form.ContactAddressQuickEntryForm.set_df_property (field_group.js:199:9)
    at setup (customer__js:267:9)
    at d (script_manager.js:30:12)
    at o (script_manager.js:109:16)
    at script_manager.js:125:5
    at Array.forEach (<anonymous>)
    at frappe.ui.form.ScriptManager.trigger (script_manager.js:122:22)
    at frappe.ui.form.ScriptManager.setup (script_manager.js:246:8)
    at frappe.ui.form.ContactAddressQuickEntryForm.setup_script_manager (quick_entry.js:134:23)
    at quick_entry.js:45:11
﻿

**Comment by mhaggag** [comment]
Can you share the `bench version` details or a screenshot of the Help -> About dialog? I can't reproduce this locally on either the `master` or the `develop` branches. The code that's throwing the error in field_group looks like this:

```
	set_df_property(fieldname, prop, value) {
		if (!fieldname) {
			return;
		}
		const field = this.get_field(fieldname);
		field.df[prop] = value; // <---- TypeError Exception thrown here because field is undefined
		field.refresh();
	}
```

That doesn't make sense to me. Can you comment out the code in `ksa_compliance/public/js/customer.js` inside the `setup` function and see if it works?

**Comment by saud-2** [comment]
![image](https://github.com/user-attachments/assets/22a56fcf-5e2b-4e54-9946-90a424ade650

> That doesn't make sense to me. Can you comment out the code in ksa_compliance/public/js/customer.js inside the setup function and see if it works?
Sorry I don't know how to do it .

**Comment by mhaggag** [comment]
This looks to be a regression or a breaking change from frappe v15.38.0, specifically this [PR](https://github.com/frappe/frappe/pull/27350). The quick entry field subclasses dialog, which somehow causes the form setup method to be called even though the form fields aren't initialized at all (because it's quick edit).

I'll open a bug on frappe and make a hot fix to check whether we are in a dialog before attempting to set the properties.
@Ahmed-M-Fawzy @khaledYusufEG I'll point to this issue in the frappe bug. I appreciate bringing the frappe team's attention to it through our contacts.

**Comment by saud-2** [comment]
Appreciate your effort. 
Thank you.

**Comment by mhaggag** [comment]
The issue opened on frappe is here: [Breaking change or regression: Quick Entry form now calls form setup without initializing fields](https://github.com/frappe/frappe/issues/27430)

**Comment by mhaggag** [comment]
I pushed a fix for this in` 0.23.1` on master. Please check and let me know.

**Comment by saud-2** [comment]
It is ok now. Thank you
